linkcheck.go

```diff
@@ -119,6 +125,20 @@ func crawlOne(req NewUrl, ch chan NewUrl, chFinished chan UrlResponse) {
 			var newUrl string
 
 			switch t.Data {
+			case "base":
+				// use the actual baseUrl set in the html file
+				ok, baseUrl := getAttr(t, "href")
+				if !ok {
+					continue
+				}
+
+				newBase, err := url.Parse(baseUrl)
+				if err != nil {
+					continue
+				}
+				base = base.ResolveReference(newBase)
+				continue
```

Comment:
Since we have a nested loop -> switch -> switch construct, suggest adding labels so the intent of what we are trying to break out of is clear.